### ISSUE TITLE
Insertion sort graphical bug

### ISSUE THREAD

**Issue by eggsontoast1672** [lunathanael/visualizing-sorting#9]
When you run insertion sort for certain array sizes, once sorting is finished, there will be a little skip in the array where it looks like a block is missing, and the right side will look off.

Steps to reproduce:

1. Switch to the `insertion-sort` branch
2. Run the command `./run.sh --elements 50`
3. Wait for the sort to run its course

For some reason, the problem only arises for certain array sizes. It's very strange.

**Comment by eggsontoast1672** [comment]
One more thing. When you quit the program after the sort finishes, the following error is printed to the console:

```
munmap_chunk(): invalid pointer
./run.sh: line 12: 16747 Aborted                 (core dumped) build/visort $@
```

I'm not sure what that's about either.

**Comment by lunathanael** [comment]
I'll take a look into it.

**Comment by lunathanael** [comment]
I found your issue, by adding an assert before your swap, it aborted with a violation. When you call std::swap with *m_insert_position, *std::next(m_insert_position), m_insert_position is already the last element, and therefore swaps with the m_end, which is undefined behavior.

**Comment by lunathanael** [comment]
Fixed with commit [2da4946](https://github.com/lunathanael/visualizing-sorting/commit/2da4946bd27e5c5e312061529849115cf75f2044).
